graphql-server/src/schemas/neo4j.ts

```diff
@@ -177,93 +182,326 @@ function buildResolvers(
         };
       },
     },
+    Query: {
+      async requests(_source: undefined, args: any) {
+        const requests = await ogm.model("Request").find({
+          where: args.where,
+          options: {
+            sort: args.options?.sort,
+          },
+          selectionSet: `{
+            igoRequestId
+            igoProjectId
+            genePanel
+            dataAnalystName
+            dataAnalystEmail
+            dataAccessEmails
+            bicAnalysis
+            investigatorEmail
+            investigatorName
+            isCmoRequest
+            labHeadEmail
+            labHeadName
+            libraryType
+            otherContactEmails
+            piEmail
+            projectManagerName
+            qcAccessEmails
+            smileRequestId
+            hasMetadataRequestMetadata {
+              importDate
+            }
+            hasSampleSamplesConnection {
+              totalCount
+            }
+          }`,
+        });
+
+        if (args.options?.sort) {
+          const sortOptions: SortOptions = args.options.sort;
+          const sortField = Object.keys(sortOptions[0])[0];
+          const sortOrder = Object.values(sortOptions[0])[0];
+
+          if (["importDate", "totalSampleCount"].includes(sortField)) {
+            sortArrayByNestedField(requests, "Request", sortField, sortOrder);
+          }
+        }
+
+        if (args.options?.limit == null) {
+          return requests;
+        }
+        return requests.slice(args.options.offset, args.options.limit);
```

Comment:
After the frontend makes the `requests` query, this GraphQL custom resolver gets called to handle that query. How it works:
- Use the Neo4j OGM to query the database. The `where` variable is passed along to handle searching. The `sort` is also passed in, but it will only handle sorting of non-nested fields.
- If the sort setting includes a nested field, we'll call `sortArrayByNestedField` to handle that.
- Finally, we'll return paginated data to handle infinite scroll. We'll only return non-paginated data when the user exports.

A similar flow applies to the `patients` query below.

The `cohorts` query follows a lite version of this flow because it currently handles sorting/filtering on the frontend (doesn't make additional requests when sorting/filtering).